File: 240409/k번째로 신기한 문자열/kth-special-string.py
N, K, T = input().split()
N, K = int(N), int(K)
words = [input() for _ in range(N)]

satisfies = []
for word in words:
    if word.startswith(T):
        satisfies.append(word)

satisfies.sort()
print(satisfies[K - 1])

File: 251230/k번째로 신기한 문자열/kth-special-string.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

int main() 
{
    int N, K;
    string T;
    cin >> N >> K >> T;

    vector<string> words(N);
    for (int i = 0; i < N; i++) 
    {
        cin >> words[i];
    }

    vector<string> satisfies;
    for (const string& word : words) 
    {
        if (word.compare(0, T.size(), T) == 0) {
            satisfies.push_back(word);
        }
    }

    sort(satisfies.begin(), satisfies.end());

    cout << satisfies[K - 1];
    return 0;
}
